lib/internal/test_runner/reporter/spec.js

```diff
@@ -36,27 +37,25 @@ const symbols = {
   'arrow:right': '\u25B6 ',
   'hyphen:minus': '\uFE63 ',
 };
-class SpecReporter extends Transform {
-  #stack = [];
-  #reported = [];
-  #indentMemo = new SafeMap();
-  #failedTests = [];
-  #cwd = process.cwd();
-
-  constructor() {
-    super({ __proto__: null, writableObjectMode: true });
-  }
 
-  #indent(nesting) {
-    let value = this.#indentMemo.get(nesting);
+module.exports = async function *specReporter(source) {
```

Comment:
This would still need to be a transform stream, API wise (can be from a generator)